packages/components/src/useSteps.js

```diff
@@ -1,13 +1,16 @@
-import { useContext, useMemo } from 'react'
-import { Context } from './context'
+import { useContext, useEffect } from 'react'
+import { Context as C } from './context'
 
-export default length => {
+export const useStepsFactory = Context => length => {
   const context = useContext(Context)
   const { register, index, step } = context
-  useMemo(() => {
+
+  useEffect(() => {
```

Comment:
I think I get the rationale here, but the reason I went with memo was that it *might* work server-side – this is definitely a "side effect" and it might not matter for static rendering

Reply:
I'd have to do some additional research into how the `useEffect` hook works when rendering on the server. I believe it should work no differently then in the browser, especially since it's not using anything browser specific? I'll circle back once I've looked into it.

Reply:
Revisiting this, I think `useEffect` is indeed the one to use for this sort of thing 👍